mesonbuild/modules/gnome.py

```diff
@@ -736,15 +736,15 @@ def generate_gir(self, state, args, kwargs):
         if kwargs.get('install_dir'):
             raise MesonException('install_dir is not supported with generate_gir(), see "install_dir_gir" and "install_dir_typelib"')
 
-        giscanner = self.interpreter.find_program_impl('g-ir-scanner')
-        gicompiler = self.interpreter.find_program_impl('g-ir-compiler')
-
```

Comment:
I think you should still keep that in the case it's not cross build, and fallback to use pkg-config.

Reply:
Is there a reason for this? Gobject-introspection has provided these paths since release, and they work not only on cross-building but also host-building.

Reply:
Also, I think you have it backward when cross-building, the above would still find host/usr/bin/g-ir-scanner, whereas using pkg-config set's the appropriate paths.

Reply:
find_program is needed in the case g-i is built as subproject. In that case ou uses override_find_program so we can use them from the build dir instead of from system.

Reply:
Ah that makes sense, although this still leads to the problem above.

If giscanner = self.interpreter.find_program_impl('g-ir-scanner') is first, then the host directory will be found first in a cross-compiled environment and the pkg-config code will never get executed.

Reply:
Right, I guess you need to check the path of the returned program, if it's within build dir or source dir then you should use it, otherwise use pkg-config.

Reply:
I updated the code, it's a bit long-winded but it should work for both scenarios.

Reply:
The first failure is due to a boost issue.

The second failure I am not sure why it's crashing. It's still using /usr/bin/g-ir-scanner, even though I check to see if the goi utils are in the source or build path. Any help would be appreciated!